src/dds/rs-dds-internal-data.cpp

```diff
@@ -0,0 +1,178 @@
+                                                                                 RS2_STREAM_COLOR );
+            for( auto & it : tmp )
+                factories.push_back( std::move( it ) );
+            factories.push_back( processing_block_factory::create_id_pbf( RS2_FORMAT_RAW16, RS2_STREAM_COLOR ) );
+            factories.push_back( { { { RS2_FORMAT_Y12I } },
+                                   { { RS2_FORMAT_Y16, RS2_STREAM_INFRARED, 1 },
+                                     { RS2_FORMAT_Y16, RS2_STREAM_INFRARED, 2 } },
+                                   []() { return std::make_shared< y12i_to_y16y16 >(); } } );
+        }
+        tmp = processing_block_factory::create_pbf_vector< uyvy_converter >(
+                               RS2_FORMAT_UYVY,
+                               target_formats( RS2_FORMAT_UYVY ),
+                               RS2_STREAM_INFRARED );
+        for( auto & it : tmp )
+            factories.push_back( std::move( it ) );
+        factories.push_back( processing_block_factory::create_id_pbf( RS2_FORMAT_Y8, RS2_STREAM_INFRARED, 1 ) );
+        factories.push_back( processing_block_factory::create_id_pbf( RS2_FORMAT_Z16, RS2_STREAM_DEPTH ) );
+        factories.push_back( { { { RS2_FORMAT_W10 } },
+                               { { RS2_FORMAT_RAW10, RS2_STREAM_INFRARED, 1 } },
+                               []() { return std::make_shared< w10_converter >( RS2_FORMAT_RAW10 ); } } );
+        factories.push_back( { { { RS2_FORMAT_W10 } },
+                               { { RS2_FORMAT_Y10BPACK, RS2_STREAM_INFRARED, 1 } },
+                               []() { return std::make_shared< w10_converter >( RS2_FORMAT_Y10BPACK ); } } );
+        factories.push_back( { { { RS2_FORMAT_Y8I } },
+                               { { RS2_FORMAT_Y8, RS2_STREAM_INFRARED, 1 },
+                                 { RS2_FORMAT_Y8, RS2_STREAM_INFRARED, 2 } },
+                               []() { return std::make_shared< y8i_to_y8y8 >(); } } );
+
+        // Motion convertion is done on the camera side.
+        // Intrinsics table is being read from the camera flash and it is too much to set here, this file is ment as a
+        // temporary solution, the camera should send all this data in a designated topic.
+        factories.push_back( { { { RS2_FORMAT_MOTION_XYZ32F, RS2_STREAM_ACCEL } },
+                               { { RS2_FORMAT_MOTION_XYZ32F, RS2_STREAM_ACCEL } },
+                               []() { return std::make_shared< identity_processing_block >(); } } );
+        factories.push_back( { { { RS2_FORMAT_MOTION_XYZ32F, RS2_STREAM_GYRO } },
+                               { { RS2_FORMAT_MOTION_XYZ32F, RS2_STREAM_GYRO } },
+                               []() { return std::make_shared< identity_processing_block >(); } } );
+    }
+
+    // For other devices empty factories will be returned, raw profiles will be used.
```

Comment:
Currently (in this PR) we only support D400 devices. I removed the exception when other devices are identified based on your previous comment in this PR.

As discussed with @Nir-Az, this will change at a future PR, supporting format types, not devices.
In that next PR we can check all the repercussions: how to support interlaced formats, what to do when rotation is needed, etc...

Reply:
Will Yair's code work with librealsense after this?

Reply:
And Z16 for the depth, of course

Reply:
> he is still using one of the old PIDs

What PID? We do not communicate a PID, just a product name and product family. It shows "Intel RealSense D555e" in the Viewer, so that's the product name...

Reply:
Whoa hold on... when did we add `product_id` to the `device_info`?!
That shouldn't be...

Reply:
>product_id is needed by viewer, if not registered there is an exception.

Then we fix the exception when we remove the `product_id` :)

Reply:
I'm ignoring this file for this code review... I expect it will be simplified and made not product-dependent...